docs(windows): Improve Windows build setup instructions

aryanjsawant opened the pull request:
## TLDR

This Pull Request updates the `CONTRIBUTING.md` to include essential setup instructions for Windows users. It addresses a persistent build hang encountered during `npm run build` and `npm run build:all` by requiring the `GEMINI_SANDBOX=true` environment variable, along with `VERBOSE=true` for better build feedback.

Reviewers should primarily focus on the new "Windows Development Setup" section and verify its clarity and accuracy.

## Dive Deeper

This change directly addresses the problem described in [Issue \#2084](https://www.google.com/search?q=https://github.com/google/gemini-cli/issues/2084), where the build process on Windows would hang indefinitely due to an early sandbox detection script, even when sandboxing was not explicitly requested (i.e., for a standard `npm run build`).

The solution involves:

1.  **Requiring `GEMINI_SANDBOX=true`:** This environment variable acts as a bypass for the problematic sandbox detection logic that was causing the hang.
2.  **Adding `VERBOSE=true`:** This provides crucial progress messages during the build process, preventing silent hangs and aiding in future debugging.
3.  **Updated `CONTRIBUTING.md`:** The documentation now includes a new "Windows Development Setup (Important\!)" section, strategically placed early in the "Development Setup and Workflow." This section guides Windows users to create a `.env` file with these variables (recommended for persistence) and provides alternative methods for setting them in various shell environments.

It's important to note that **this PR primarily fixes the *build hang* on Windows and updates the CONTRIBUTING.md accordingly.** It does **not** address the existing unit test failures (specifically in `packages/core/src/utils/memoryDiscovery.test.ts`) that occur after a successful build. Those will need to be addressed in a subsequent contribution to achieve a fully passing `npm run preflight`.

## Reviewer Test Plan

Reviewers can validate this change by performing the following steps on a **Windows machine**:

1.  **Clean Clone:** Start with a fresh clone of the `gemini-cli` repository.
2.  **Follow New Instructions:** Apply the instructions in the new "Windows Development Setup (Important\!)" section in `CONTRIBUTING.md` (e.g., create the `.env` file with `GEMINI_SANDBOX=true` and `VERBOSE=true`).
3.  **Install Dependencies:** Run `npm install`.
4.  **Run Build:** Execute `npm run build`.
      * **Expected Behavior:** The build should now complete successfully without hanging, and you should see verbose output during the process.
5.  **Run Sandbox Build (Optional, if Docker/Podman is available):** Execute `npm run build:all`.
      * **Expected Behavior:** This should also complete successfully without hanging, building the sandbox container.
6.  **Run Preflight:** Execute `npm run preflight`.
      * **Expected Behavior:** The build steps of `preflight` should complete without hanging. The tests will still likely show failures (e.g., in `memoryDiscovery.test.ts`), but the initial hang should be resolved.

For **Linux/macOS users**, please verify that these documentation changes do not negatively impact the existing build or development setup.

## Testing Matrix

|            | 🍏  | 🪟  | 🐧  |
| ---------- | --- | --- | --- |
| `npm run`  | ❓  | ✅  | ❓  |
| `npx`      | ❓  | ✅  | ❓  |
| `Docker`   | ❓  | ✅  | ❓  |
| `Podman`   | ❓  | -  | -   |
| `Seatbelt` | ❓  | -   | -   |

## Linked issues / bugs

Fixes \#2084

-----

DinoChiesa commented:
The portion in the modified CONTRIBUTING document, that says 

> Some environments (e.g., PowerShell, CMD) may not automatically load `.env` files. Consider using [`dotenv-cli`](https://github.com/entropitor/dotenv-cli) or ensure your shell is configured to source `.env`.

I suggest you amend that to add: 

> If you have dotenv-cli , then you can use this command to build:
>
>        dotenv npm run build
>
> or
>
>        dotenv npm run bundle 
>

aryanjsawant commented:
In the new commit,

I've removed the line:
Some environments (e.g., PowerShell, CMD) may not automatically load .env files. Consider using [dotenv-cli](https://github.com/entropitor/dotenv-cli) or ensure your shell is configured to source .env.

and added the suggested:
If you have dotenv-cli , then you can use this command to build:
   dotenv npm run build
or

   dotenv npm run bundle


in proper markdown syntax

galz10 commented:
Hi @aryanjsawant, thanks for submitting this, It looks like the original issue has already been fixed, so we're closing this PR, but if there is something you think I've missed please reopen and ping me
